src/main/java/com/tcg/lwjgllearning/graphics/g3d/mesh/AbstractMesh.java

```diff
@@ -0,0 +1,39 @@
+package com.tcg.lwjgllearning.graphics.g3d.mesh;
+
+import java.util.HashMap;
+
+import com.tcg.lwjgllearning.graphics.Drawable;
+import com.tcg.lwjgllearning.math.Quaternion;
+import com.tcg.lwjgllearning.math.Transform3D;
+import com.tcg.lwjgllearning.math.Vector3;
+import com.tcg.lwjgllearning.utils.Disposable;
+
+import static org.lwjgl.opengl.GL15.*;
+import static org.lwjgl.opengl.GL20.*;
+
+public abstract class AbstractMesh extends Transform3D implements Disposable, Drawable {
+    
+    protected HashMap<AbstractAttribute, Integer> attributeBufferMap;
+
+    public abstract void draw();
+    protected abstract void activate();
+    protected abstract void deactivate();
+
+    protected AbstractMesh(Vector3 position, Quaternion rotation, Vector3 scale) {
+        super(position, rotation, scale);
+    }
+
+    protected void activateAttribute3f(AbstractAttribute attribute, int attributeLocation) {
```

Comment:
These functions seem to only be used by unused functions, can you explain their purpose?

Reply:
The unused functions that use those are intended to be used when you need particular attribute(s) to be used in a draw call with a shader and/or vao other than those associated with the mesh containing said attributes. For example, when creating a particular light's depth buffer for shadow mapping, a very simple shader and a vao other than the mesh's will be used, but the mesh's position attribute will be needed.